### PR TITLE
feat: Docker support for the celestia-prover

### PR DESCRIPTION
**Pull request by cmwaters**
Closes: https://github.com/celestiaorg/celestia-zkevm-ibc-demo/issues/54

This PR creates a docker file for running the celestia-prover instance and adds it to the docker compose file

It also changes the ghcr images to pull from "ghcr.io/celestia-zkevm-ibc-demo/simapp"

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -108,14 +109,22 @@ proto-format:
...
+	$(DOCKER) build -t $(SIMAPP_GHCR_REPO) -f docker/simapp.Dockerfile .
 .PHONY: build-simapp-docker
 
 ## publish-simapp-docker: Publish the simapp docker image to GHCR. Requires Docker and authentication.
 publish-simapp-docker:
-	$(DOCKER) push $(GHCR_REPO)
+	$(DOCKER) push $(SIMAPP_GHCR_REPO)
 .PHONY: publish-simapp-docker
 
+build-celestia-prover-docker:
```
Comment (rootulp) [inline Makefile:120]:
```suggestion
## build-celestia-prover-docker: Build the celestia prover docker image from the current branch. Requires docker.
build-celestia-prover-docker:
```

File: Makefile [hunk 2]
```diff
@@ -108,14 +109,22 @@ proto-format:
...
+	$(DOCKER) build -t $(CELESTIA_PROVER_GHCR_REPO) -f docker/celestia_prover.Dockerfile .
+.PHONY: build-celestia-prover-docker
+
+publish-celestia-prover-docker:
```
Comment (rootulp) [inline Makefile:124]:
```suggestion
## publish-celestia-prover-docker: Publish the celestia prover docker image from the current branch. Requires docker.
publish-celestia-prover-docker:
```

**Review by rootulp — COMMENTED** [review]
I can't pull images from ghcr.io/celestiaorg/celestia-zkevm-ibc-demo/simapp and I can't push to that repo b/c insufficient permissions

```
$ make publish-simapp-docker
/opt/homebrew/bin/docker push ghcr.io/celestiaorg/celestia-zkevm-ibc-demo/simapp
Using default tag: latest
The push refers to repository [ghcr.io/celestiaorg/celestia-zkevm-ibc-demo/simapp]
5f70bf18a086: Preparing
7276f72e750d: Preparing
c714b8be4359: Preparing
651d9022c234: Preparing
denied: permission_denied: write_package
make: *** [publish-simapp-docker] Error 1
```

**Comment by cmwaters** [comment]
@rootulp you should have access now. I've linked the packages to the repository

**Review by rootulp — APPROVED** [review]
LGTM. `docker start` fetched the image and I can run it

```
$ grpcurl -plaintext localhost:50051 describe celestia.prover.v1.Prover
celestia.prover.v1.Prover is a service:
service Prover {
  rpc ProveMembership ( .celestia.prover.v1.ProveMembershipRequest ) returns ( .celestia.prover.v1.ProveMembershipResponse );
  rpc ProveStateTransition ( .celestia.prover.v1.ProveStateTransitionRequest ) returns ( .celestia.prover.v1.ProveStateTransitionResponse );
}
```

In a follow up PR we could add a command to make and publish _all_ the Docker images. Something like `make docker` builds and publishes both images

**Comment by rootulp** [comment]
Oof go.mod merge conflict. I'll fix and push

**Comment by cmwaters** [comment]
Yeah I think it would be nice to add a workflow like in other repos to push new images when there's been changes to the relevant directories

**Pull request merged**